File: src/main/java/ch/uzh/ifi/access/repository/SubmissionFileRepository.java
package ch.uzh.ifi.access.repository;

import ch.uzh.ifi.access.model.SubmissionFile;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface SubmissionFileRepository extends JpaRepository<SubmissionFile, Long> {
    Optional<SubmissionFile> findTopByTaskFile_IdAndSubmission_UserIdOrderByIdDesc(Long taskFileId, String userId);
}

File: src/main/kotlin/ch/uzh/ifi/access/repository/SubmissionFileRepository.kt
package ch.uzh.ifi.access.repository

import ch.uzh.ifi.access.model.SubmissionFile
import org.springframework.data.jpa.repository.JpaRepository

interface SubmissionFileRepository : JpaRepository<SubmissionFile?, Long?> {
    fun findTopByProblemFile_IdAndSubmission_UserIdOrderByIdDesc(
        problemFileId: Long?,
        userId: String?
    ): SubmissionFile?
}